model_management/mm_run_minimal/emodels_dir/singlecell-optimization/mechanisms/StochKv3.mod

```diff
@@ -302,46 +320,56 @@ ENDVERBATIM
 FUNCTION urand() {
 
 VERBATIM
-    double value;
+    double value = 0.0;
     if( usingR123 ) {
         value = nrnran123_dblpick((nrnran123_State*)_p_rng);
     } else if (_p_rng) {
-        value = nrn_random_pick(_p_rng);
+#ifndef CORENEURON_BUILD
+        value = nrn_random_pick(RANDCAST _p_rng);
+#endif
     } else {
-        value = 0.5;
+        // see BBPBGLIB-972
+        value = 0.0;
     }
```

Comment:
> "The changes to mod files like StochKv3 are all technical. There is no change in intrinsic behaviour / results "

Just to remain "true" to the above quote: this change to `value` variable is the one something that we were a bit unsure about. In the JIRA [linked issue](https://bbpteam.epfl.ch/project/issues/browse/BBPBGLIB-972) we explained why this was necessary and how this was the issue in load-balancing / ExperimentalMechComplex usage with NEURON.

We have ran simulations and results are same and now this change is in production. But if you really want to keep exact old code, you can set the value back to `0.5`.

Reply:
Just to add some more information if someone will revisit this in the future: 

```console
FUNCTION urand() {VERBATIM
    double value;
    if( usingR123 ) {
        value = nrnran123_dblpick((nrnran123_State*)_p_rng);
    } else if (_p_rng) {
#ifndef CORENEURON_BUILD
        value = nrn_random_pick(RANDCAST _p_rng);
#endif
    } else {
        
        value = 0.5;
    }
    _lurand = value;
ENDVERBATIM
}
```

In the above `urand()` function we are picking a random number. In all practical usage, we would have an RNG stream setup either using Random123 or MCellRan4. The last `else` block returns a value `0.5 (or 0)` when a user hasn't set up the RNG variable i.e. StochKV is not properly instantiated. This won't be the case if someone is really using StochKV.

So I would say let's ignore my initial comment about this change. Using the new MOD file is perfectly OK.

Reply:
Thanks for the detailed explanation @pramodk!